examples/Cpp/PrimitiveRestart/Example.cpp

```diff
@@ -0,0 +1,177 @@
+/*
+ * Example.cpp (Example_PrimitiveRestart)
+ *
+ * This file is part of the "LLGL" project (Copyright (c) 2015-2019 by Lukas Hermanns)
+ * See "LICENSE.txt" for license information.
+ */
+
+#include <ExampleBase.h>
+#include <vector>
+
+class Example_PrimitiveRestart : public ExampleBase
+{
+    LLGL::ShaderProgram*    shaderProgram       = nullptr;
+    LLGL::PipelineState*    pipeline            = nullptr;
+    LLGL::Buffer*           vertexBuffer        = nullptr;
+    LLGL::Buffer*           indexBuffer16       = nullptr;
+    LLGL::Buffer*           indexBuffer32       = nullptr;
+
+    size_t                  indexCount16        = 0;
+    size_t                  indexCount32        = 0;
+
+public:
+
+    Example_PrimitiveRestart() :
+    ExampleBase { L"LLGL Example: PrimitiveRestart" }
+    {
+        auto vertexFormats = CreateBuffers();
+        shaderProgram = LoadStandardShaderProgram(vertexFormats);
+        CreatePipeline();
+    }
+
+private:
+
+    // Vertex data structure
+    struct Vertex
+    {
+        Gs::Vector2f        position;
+        LLGL::ColorRGBAub   color;
+    };
+
+    template <typename T>
+    static void addSquare(float centerX, float centerY, float size, std::vector<Vertex>& vertices, std::vector<T>& indices)
+    {
+        float left = centerX - size / 2.0;
+        float right = centerX + size / 2.0;
+        float top = centerY + size / 2.0;
+        float bottom = centerY - size / 2.0;
+
+        size_t startIndex = vertices.size();
```

Comment:
`startIndex` should be of type `T` because [MSVC 2019 Win64 is complaining](https://ci.appveyor.com/project/LukasBanana/llgl/builds/34031207/job/50f2t2pm0u3kkte6#L204) about implicit truncation from `size_t` to `T`. I suggest something like this:
```cpp
const T startIndex = static_cast<T>(vertices.size());
```
The conversion between `size_t` and `uint32_t` can be a bit of a pain in LLGL, but 32 bit unsigned integers are all we get (most of the time) from rendering APIs up until Vulkan and D3D12.